### PR TITLE
I210 automatic leaderboard upload at the end of training

### PR DESCRIPTION
**Pull request by nathanlct**
## Pull request information

- **Status**: ready to merge
- **Kind of changes**: improved workflow

## Description

Automatically generate graphs and upload them to leaderboard when training on i210 with a new command-line argument `--upload_graphs <submitter_name> <strategy_name>`

Example command:

```
python flow/examples/train.py multiagent_i210 i210_ppo --num_iterations 40 --num_rollouts 20 \
--checkpoint_freq 10 --use_s3 --num_cpus 20 --algorithm PPO --rl_trainer rllib --grid_search \
--upload_graphs Nathan 10%_ppo_default
```

### CODE REVIEW

File: examples/train.py [hunk 1]
```diff
@@ -185,7 +191,7 @@ def setup_exps_rllib(flow_params,
         config["train_batch_size"] = horizon * n_rollouts
         config["gamma"] = 0.995  # discount rate
         config["use_gae"] = True
-        config["no_done_at_end"] = False
+        config["no_done_at_end"] = True
```
Comment (eugenevinitsky) [inline examples/train.py:194]: Is this needed?

Comment (nathanlct) [reply examples/train.py:194]: I had an error when setting it to true. It's rare but the bug's still there

File: examples/train.py [hunk 2]
```diff
@@ -374,6 +382,37 @@ def trial_str_creator(trial):
...
+        for (dirpath, _, _) in os.walk(os.path.expanduser("~/ray_results")):
+            if "checkpoint_{}".format(flags.checkpoint_freq) in dirpath \
+                and dirpath.split('/')[-3] == flags.exp_title:
+                checkpoint_path = os.path.dirname(dirpath)
+                checkpoint_number = -1
+                for name in os.listdir(checkpoint_path):
+                    if name.startswith('checkpoint'):
+                        cp = int(name.split('_')[1])
+                        checkpoint_number = max(checkpoint_number, cp)
+                break
```
Comment (nathanlct) [inline examples/train.py:399]: Probably remove this break (and indent what follows) to generate plots for all the grid-search values -- although that would upload a lot of curves on the leaderboard each time, I'm not sure if that's a problem

Comment (eugenevinitsky) [reply examples/train.py:399]: It's not a problem as long as we name the results with the grid search values concatenated in some way

File: examples/train.py (hunk 1 above)
Comment (eugenevinitsky) [reply examples/train.py:194]: Argh, okay.

File: examples/train.py (hunk 2 above)
Comment (eugenevinitsky) [reply examples/train.py:399]: It's just an extra half hour per grid point which is fine.

Comment (nathanlct) [reply examples/train.py]: Alright it's fixed

**Comment by eugenevinitsky** [comment]
@nathanlct if you resolve the conflicts we can get this merged

**Comment by eugenevinitsky** [comment]
It's probably also failing flake8 / pydoc

**Comment by eugenevinitsky** [comment]
It's probably also failing flake8 / pydoc

File: examples/train.py [hunk 3]
```diff
@@ -275,8 +282,7 @@ def on_episode_step(info):
         av_speed = np.mean([speed for speed in env.k.vehicle.get_speed(rl_ids) if speed >= 0])
         if not np.isnan(av_speed):
             episode.user_data["avg_speed_avs"].append(av_speed)
-        episode.user_data["avg_mpg"].append(miles_per_gallon(env, veh_ids, gain=1.0))
-        episode.user_data["avg_mpj"].append(miles_per_megajoule(env, veh_ids, gain=1.0))
```
Comment (eugenevinitsky) [inline examples/train.py:279]: why is this deleted?
